tests/acceptance/products/PageShortcodeProductCest.php

```diff
@@ -263,6 +263,58 @@ public function testProductShortcodeInVisualEditorWithInvalidDiscountCodeParamet
 		$I->see('The coupon is not valid.');
 	}
 
+	/**
+	 * Test the Product shortcode opens the ConvertKit Product in the same window instead
+	 * of a modal when the Disable modal on mobile option is enabled.
+	 *
+	 * @since   2.4.1
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function testProductShortcodeWithDisableModalOnMobileParameterEnabled(AcceptanceTester $I)
+	{
+		// Setup ConvertKit Plugin with no default form specified.
+		$I->setupConvertKitPluginNoDefaultForms($I);
+		$I->setupConvertKitPluginResources($I);
+
+		// Add a Page using the Classic Editor.
+		$I->addClassicEditorPage($I, 'page', 'ConvertKit: Page: Product: Shortcode: Disable Modal on Mobile');
+
+		// Add shortcode to Page, setting the Product setting to the value specified in the .env file.
+		$I->addVisualEditorShortcode(
+			$I,
+			'ConvertKit Product',
+			[
+				'product'                 => [ 'select', $_ENV['CONVERTKIT_API_PRODUCT_NAME'] ],
+				'disable_modal_on_mobile' => [ 'toggle', 'Yes' ],
+			],
+			'[convertkit_product product="' . $_ENV['CONVERTKIT_API_PRODUCT_ID'] . '" text="Buy my product" disable_modal_on_mobile="1"]'
+		);
+
+		// Publish and view the Page on the frontend site.
+		$I->publishAndViewClassicEditorPage($I);
+
+		// Get Page URL.
+		$url = $I->grabFromCurrentUrl();
+
+		// Change user agent to a mobile user agent.
+		$I->changeUserAgent($_ENV['TEST_SITE_HTTP_USER_AGENT_MOBILE']);
+
+		// Load page.
+		$I->amOnPage($url);
+
+		// Confirm that the shortcode displays without the data-commerce attribute.
+		$I->seeElementInDOM('.convertkit-product a');
+		$I->dontSeeElementInDOM('.convertkit-product a[data-commerce]');
+
+		// Confirm that clicking the button opens the URL in the same browser tab, and not a modal.
+		$I->click('.convertkit-product a');
+		$I->waitForElementVisible('body[data-template]');
+
+		// Change user agent back, as it persists through tests.
+		$I->changeUserAgent($_ENV['TEST_SITE_HTTP_USER_AGENT']);
```

Comment:
This might be the wrong question, but does a failed assertion throw an exception? Or will this code run even if there's a failed assertion before it? If so, is there a way to put cleanup outside the test, so it always runs? Pardon my ignorance.

Reply:
If `waitForElementVisible` doesn't find the expected HTML element (`body[data-template]` being visible if we're viewing the ConvertKit Product page directly and not in a modal), the test would fail.

If the `changeUserAgent` call fails i.e. chromedriver cannot be restarted with the given user-agent, later tests would fail as chromedriver won't be running.  It's not critical that the user agent be changed back for other tests, just good practice.